src/meepgeom.cpp

```diff
@@ -1842,11 +1842,11 @@ static int geom_boxes_intersect(const geom_box *b1, const geom_box *b2) {
   /* true if the x, y, and z ranges all intersect. */
   return (
       (BETWEEN(b1->low.x, b2->low.x, b2->high.x) || BETWEEN(b1->high.x, b2->low.x, b2->high.x) ||
-       BETWEEN(b2->low.x, b1->low.x, b1->high.x)) &&
+       BETWEEN(b2->low.x, b1->low.x, b1->high.x) || BETWEEN(b2->high.x, b1->low.x, b1->high.x)) &&
```

Comment:
Isn't this redundant with the other three tests?

Reply:
That is, there are only four ways two intervals `b1 = (...)` and `b2 = [...]` can overlap:

1.   `[ ...(...]...)` : `BETWEEN(b1->low.x, b2->low.x, b2->high.x) == true`
2.   `(...[...)...]`: `BETWEEN(b1->high.x, b2->low.x, b2->high.x) == true`
3. `[...(...)...]`: both of the above checks are`true`
4. `(...[...]...)`: `BETWEEN(b2->low.x, b1->low.x, b1->high.x) == true`

So, there is no need to also check `BETWEEN(b2->high.x, b1->low.x, b1->high.x)`

Reply:
Yes, you're right: the extra check is redundant and thus unnecessary.